pkgs/misc/uboot/default.nix

```diff
@@ -179,50 +181,113 @@ in {
     filesToInstall = ["u-boot-dtb.img" "SPL"];
   };
 
-  # Flashing instructions:
-  # dd if=bl1.bin.hardkernel of=<device> conv=fsync bs=1 count=442
-  # dd if=bl1.bin.hardkernel of=<device> conv=fsync bs=512 skip=1 seek=1
-  # dd if=u-boot.gxbb of=<device> conv=fsync bs=512 seek=97
-  ubootOdroidC2 = let
-    firmwareBlobs = fetchFromGitHub {
-      owner = "armbian";
-      repo = "odroidc2-blobs";
-      rev = "47c5aac4bcac6f067cebe76e41fb9924d45b429c";
-      sha256 = "1ns0a130yxnxysia8c3q2fgyjp9k0nkr689dxk88qh2vnibgchnp";
-      meta.license = lib.licenses.unfreeRedistributableFirmware;
-    };
-  in buildUBoot {
+  ubootOdroidC2 = buildUBoot {
     defconfig = "odroid-c2_defconfig";
-    extraMeta.platforms = ["aarch64-linux"];
-    filesToInstall = ["u-boot.bin" "u-boot.gxbb" "${firmwareBlobs}/bl1.bin.hardkernel"];
+
     postBuild = ''
-      # BL301 image needs at least 64 bytes of padding after it to place
-      # signing headers (with amlbootsig)
-      truncate -s 64 bl301.padding.bin
-      cat '${firmwareBlobs}/gxb/bl301.bin' bl301.padding.bin > bl301.padded.bin
-      # The downstream fip_create tool adds a custom TOC entry with UUID
-      # AABBCCDD-ABCD-EFEF-ABCD-12345678ABCD for the BL301 image. It turns out
-      # that the firmware blob does not actually care about UUIDs, only the
-      # order the images appear in the file. Because fiptool does not know
-      # about the BL301 UUID, we would have to use the --blob option, which adds
-      # the image to the end of the file, causing the boot to fail. Instead, we
-      # take advantage of the fact that UUIDs are ignored and just put the
-      # images in the right order with the wrong UUIDs. In the command below,
-      # --tb-fw is really --scp-fw and --scp-fw is the BL301 image.
-      #
-      # See https://github.com/afaerber/meson-tools/issues/3 for more
-      # information.
-      '${buildPackages.armTrustedFirmwareTools}/bin/fiptool' create \
-        --align 0x4000 \
-        --tb-fw '${firmwareBlobs}/gxb/bl30.bin' \
-        --scp-fw bl301.padded.bin \
-        --soc-fw '${armTrustedFirmwareS905}/bl31.bin' \
-        --nt-fw u-boot.bin \
+      ${firmwareOdroidC2}/fip_create \
+        --bl30 ${firmwareOdroidC2}/bl30.bin \
+        --bl301 ${firmwareOdroidC2}/bl301.bin \
+        --bl31 ${firmwareOdroidC2}/bl31.bin \
+        --bl33 u-boot.bin \
         fip.bin
-      cat '${firmwareBlobs}/gxb/bl2.package' fip.bin > boot_new.bin
-      '${buildPackages.meson-tools}/bin/amlbootsig' boot_new.bin u-boot.img
-      dd if=u-boot.img of=u-boot.gxbb bs=512 skip=96
+      ${firmwareOdroidC2}/fip_create --dump fip.bin
+
+      cat ${firmwareOdroidC2}/bl2.package fip.bin > boot_new.bin
+      ${firmwareOdroidC2}/aml_encrypt_gxb --bootsig \
```

Comment:
Same here. `amlbootsig` is an open source re-implementation of this tool.

Reply:
reverted back to amlbootsig